plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1062,57 +1067,60 @@ bool chain_plugin::recover_reversible_blocks( const fc::path& db_dir, uint32_t c
 
    ilog( "Reconstructing '${reversible_dir}' from backed up reversible directory", ("reversible_dir", reversible_dir) );
 
-   chainbase::database  old_reversible( backup_dir, database::read_only, 0, true );
-   chainbase::database  new_reversible( reversible_dir, database::read_write, cache_size );
-   std::fstream         reversible_blocks;
-   reversible_blocks.open( (reversible_dir.parent_path() / std::string("portable-reversible-blocks-").append( now ) ).generic_string().c_str(),
+   try {
```

Comment:
I think we should change this to:

fc::optional<chainbase::databes> old_reversible;
try {
   old_reversible = chainbase::database( backup_dir, database::read_only, 0, true );
} catch(const std::runtime_error&) {
   ilog( "Did not recover any reversible blocks since reversible database incompatible");
   // since we are allowing for dirty, it must be incompatible
   return true;
}

And keep the rest as it was, so that we do not scoop up any other exceptions that we have not analyzed.

Reply:
I like that.  C++17 has an optional, can we use C++17 features?

Reply:
We can use C++17 features. However, fc has an optional. Please use that until we move from fc option to C++17 optional.